src/html/mod.rs

```diff
@@ -45,6 +45,54 @@ pub trait Component: Sized + 'static {
     fn destroy(&mut self) {} // TODO Replace with `Drop`
 }
 
+/// A pure component does not hold state, only providing rendering capability to a collection of properties.
+///
+/// PureComponents should be used instead of Components if the Component in question does not mutate
+/// the state passed to it via props and if it does not own services or connections to agents.
+pub trait PureComponent: Properties + PartialEq + Sized + 'static {
```

Comment:
When trait specialization arrives in a year or three, it might be possible to remove this `PartialEq` bound, and have blanket impls of `Component` for both `T: PureComponent` and `T: PureComponent + PartialEq`. The component would effectively memoize results like it does currently if the struct in question implements `PartialEq`, or just return `true` from `change` if it doesn't impl `PartialEq`.